### ISSUE TITLE
explicit tabindex settings make lightbox inaccessible for screenreaders/keyboard nav

### ISSUE THREAD

**Issue by adrienne** [biati-digital/glightbox#218]
**Describe the bug**
The lightbox next, previous, and close buttons have explicit positive `tabindex` values, which is a *huge* no-no for accessibility as it interferes with the default tabbing order. Links and buttons *do not need a tabindex value at all* but if it is set, it should be set to `tabindex="0"` to preserve logical document flow. Source ordering in the document should be used to tweak the actual tab order. 

**Are you able to reproduce the bug in the demo site**
Yes; screencap attached showing explicit tabindex values
![image](https://user-images.githubusercontent.com/89705/111673303-116d6600-87d8-11eb-889f-4bffb9d70e3a.png)

**To Reproduce**
1. https://biati-digital.github.io/glightbox/
2. Click the first image in the first gallery
3. Attempt to navigate via tab
4. You will notice that the focus jumps all over -- from the Next button to the *address bar* then back to the document, and finally to Previous and Closed

**Expected behavior**
Tab focus should be placed in logical order with the document flow, rather than jumping all over. 

If i have time i will try to fork and tweak this.

**Comment by frankie-tech** [comment]
Thanks for filing a bug report!

I made the changes you suggested, and I'll make the pull request now.

**Comment by biati-digital** [comment]
Hi @adrienne could you please elaborate more on "Jumping all over"? @frankie-tech actually the change in your PR will break the tab navigation because currently it's handled by the keyboard navigation code and it depends on the tab index value to change focus in the specified order.

Tab is handled with javascript because Safari has a tabindex issue as button elements won't gain focus when pressing tab (as mentioned here [https://codepen.io/alexandtheweb/pen/raQrxw](https://codepen.io/alexandtheweb/pen/raQrxw)) and also pressing tab behaves differently depending on the browser, for example, Chrome and Firefox.

**Comment by frankie-tech** [comment]
I was looking into the "jumping all over" bit, and I'm wondering if it has something to do with these guidelines here: https://bitsofco.de/accessible-modal-dialog/

- 1 Markup the Dialog and Dialog Overlay Appropriately
- 2 On Dialog Open, Set Focus
- 3 On Dialog Close, Return Focus to the Last Focused Element
- 4 While Open, Prevent Mouse Clicks Outside the Dialog
- 5 While Open, Prevent Tabbing to Outside the Dialog
- 6 While Open?
- 7 Allow the ESC Key to Close the Dialog

I think it is number 5 that is specifically what is being referenced by "jumping all over".

**Comment by biati-digital** [comment]
I was thinking the same, I already did some changes to prevent tabbing to outside elements, I'll leave `tabindex="0"` and use custom data attributes to customize the order, make it behave the same in all browsers and hopefully can be useful for the next version and the inclusion of plugins support. Keyboard navigation can be easily disabled if required.

In the future we can review the current functionality and improve it if required. 

I'll publish an update today.

**Comment by biati-digital** [comment]
The update is ready to download, using tab to navigate the Lightbox controls should work correctly without the "jumping all over" issue.
